src/selector.rs

```diff
@@ -0,0 +1,24 @@
+//! Abstract Faiss ID selector
+use faiss_sys::*;
+
+/// Abstraction over IDSelectorRange and IDSelectorBatch
+#[derive(Debug)]
+pub struct IdSelector {
+    pub inner: *mut FaissIDSelector,
+}
+
+impl IdSelector {
+    /// Create new range selector
+    pub fn range(min: idx_t, max: idx_t) -> IdSelector {
+        let mut sel = FaissIDSelector_H {_unused: []};
```

Comment:
In C, one would declare a pointer variable and pass a pointer to that one to the function:

```c
FaissIDSelector* p_sel = NULL;
int e = faiss_IDSelectorRange_new(&p_sel, 0, 10);
// handle e
```

The equivalent in Rust is:

```rust
let mut p_sel = ptr::null_mut();
faiss_try!(faiss_IDSelectorRange_new(&mut p_sel, 0, 10));
```

A significant portion of the types declared in the bindings are opaque types. They are not supposed to be constructed on the Rust side of the program. In case of doubt, you may follow the other abstractions in this crate (`IndexImpl`).